packages/providers/src/theme.tsx

```diff
@@ -56,48 +56,102 @@ type ThemeContextType = {
 const ThemeContext = React.createContext<ThemeContextType | undefined>(undefined);
 ThemeContext.displayName = 'ThemeContext';
 
-const prefersLightMQ = '(prefers-color-scheme: light)';
+const PREFERS_LIGHT_MQ = '(prefers-color-scheme: light)';
+
+const THEME_KEY = 'myst:theme';
+
+/**
+ * A blocking element that runs on the client before hydration to update the <html> preferred class
+ * This ensures that the hydrated state matches the non-hydrated state (by updating the DOM on the
+ * client between SSR on the server and hydration on the client)
+ */
+export function BlockingThemeLoader({ useLocalStorage }: { useLocalStorage: boolean }) {
```

Comment:
I think that we should be making the providers code a lot _smaller_ and removing this logic/components, and bringing it into the site instead.

For example, when instantiating the providers, put the loading/setting functions in at that stage (allowing different themes to make different choices). E.g. setting a cookie, modifying the dom, etc. are part of the site implementation, where the providers are just the interface to the state.